msal_extensions/cache_lock.py

```diff
@@ -12,6 +13,8 @@ class CrossPlatLock(object):
     """
     def __init__(self, lockfile_path):
         self._lockpath = lockfile_path
+        # Support for passing through arguments to the open syscall was added in v1.4.0
```

Comment:
Nice catch. In hindsight, we should have declare the dependency of `portalocker>=1.4.0` since day 1.

Are you saying "portalocker>=1.4.0 would cut off the current Debian Stable and Ubuntu before 19.10"? Why so?

If that is really the case, I'm fine with your proposed run-time detection, for now. (But if in the future we would hear bug reports saying [the default `buffering=-1` behavior](https://docs.python.org/3/library/functions.html#open) causes trouble, we would then switch to have a hard dependency on portalocker>=1.4.0

Reply:
https://packages.ubuntu.com/search?keywords=python3-portalocker&searchon=names&suite=all&section=all
https://packages.debian.org/search?keywords=python3-portalocker&searchon=names&suite=all&section=all